src/components/rules-list-page/SyncRulesPage.jsx

```diff
@@ -201,6 +206,37 @@ class SyncRulesPage extends React.Component {
         }
     };
 
+    openSharingSettings = object => {
+        this.setState({
+            sharingSettingsObject: {
+                object,
+                meta: { allowPublicAccess: true, allowExternalAccess: false },
+            },
+        });
+    };
+
+    closeSharingSettings = () => {
+        this.setState({ sharingSettingsObject: null });
+    };
+
+    verifyUserHasPermissions = (d2, rule) => {
+        const { isAdmin } = this.props;
+        const { id: userId, userGroups } = this.state.userInfo;
+        const { publicAccess = "--------", userAccesses = [], userGroupAccesses = [] } = rule;
+
+        return (
```

Comment:
That's a lot of logic for a component, maybe method in syncRule?

Reply:
Sounds good